packages/plugin-npm-cli/sources/commands/npm/login.ts

```diff
@@ -0,0 +1,162 @@
+import {Configuration, MessageName}                     from '@berry/core';
+import {PluginConfiguration, StreamReport}              from '@berry/core';
+import {npmConfigUtils, npmHttpUtils}                   from '@berry/plugin-npm';
+import inquirer                                         from 'inquirer';
+import {Readable, Writable}                             from 'stream';
+import {URL}                                            from 'url';
+
+const SUCCESS_MESSAGE = `Successfully logged in`;
+const ERROR_MESSAGE = `Invalid Authentication`;
+
+// eslint-disable-next-line arca/no-default-export
+export default (clipanion: any, pluginConfiguration: PluginConfiguration) => clipanion
+
+  .command(`npm login [-r,--registry REGISTRY]`)
+  .categorize(`Npm-related commands`)
+  .describe(`login to registry`)
+
+  .detail(`
+    This command will ask you for your registry username, email, password, and One-time password (when it applies).
```

Comment:
```suggestion
    This command will ask you for your username, email, password, and 2FA One-Time Password (when it applies). It will then modify your local configuration (in your home folder) to reference the new tokens thus generated.

    Adding the \`-s,--scope\` flag will cause the authentication to be done against whatever registry is configured for the associated scope (see also \`npmScopes\`).
```

Btw, is asking for the email required? This is something I never understood from the npm authentication scheme - why do they care about my email if they already have my username? Especially since the email is public data through the registry ...

Reply:
Hi !! sorry late to the party :( . At least Verdaccio it is not used at all, but I've been thinking perhaps at least to let it go through, but, I haven't found any scenario where is useful or any user has not even required to use it in the last two years.